python/lsst/ctrl/bps/report.py

```diff
@@ -127,17 +143,97 @@ def report(wms_service, run_id, user, hist_days, pass_thru, is_global=False, ret
             print(
                 f"No records found for job id '{run_id}'. "
-                f"Hints: Double check id, retry with a larger --hist value (currently: {hist_days}), "
+                f"Hints: Double check id, retry with a larger --hist value (currently: {hist}), "
                 "and/or use --global to search all job queues."
             )
     else:
+        runs, messages = retrieve_report(
+            wms_service,
+            user=user,
+            hist=hist,
+            pass_thru=pass_thru,
+            is_global=is_global,
+        )
         for run in runs:
             run_brief.add(run, use_global_id=is_global)
         run_brief.sort("ID")
         print(run_brief)
-    if message:
-        print(message)
+
+    if messages:
+        print("\n".join(messages))
         print("\n")
+
+
+def retrieve_report(
+    wms_service: str,
+    *,
+    run_id: str | None = None,
+    user: str | None = None,
+    hist: float | None = None,
+    pass_thru: str | None = None,
+    is_global: bool = False,
+    postprocessors: Sequence[Callable[[WmsRunReport], None]] | None = None,
+) -> tuple[list[WmsRunReport], list[str]]:
+    """Retrieve summary of jobs submitted for execution.
+
+    Parameters
+    ----------
+    wms_service : `str`
+        Name of the WMS service class.
+    run_id : `str`, optional
+        A run id the report will be restricted to.
+    user : `str`, optional
+        A username the report will be restricted to.
+    hist : int, optional
+        Include runs from the given number of past days.
+    pass_thru : `str`, optional
+        A string to pass directly to the WMS service class.
+    is_global : `bool`, optional
+        If set, all available job queues will be queried for job information.
+        Defaults to False which means that only a local job queue will be
+        queried for information.
+
+        Only applicable in the context of a WMS using distributed job queues
+        (e.g., HTCondor).
+    postprocessors : `collections.abc.Sequence` [callable], optional
+        List of functions for "massaging" reports returned by the plugin. Each
+        function must take one positional argument:
+
+        - ``report``: run report (`lsst.ctrl.bps.WmsRunReport`)
+
+        If None (default), each run report returned by the plugin (if any)
+        will be returned as is.
+
+    Returns
+    -------
+    reports : `list` [`WmsRunReport`]
+        Run reports satisfying the search criteria.
+    messages : `list` [`str`]
+        Errors that happened during report retrieval and/or processing.
+        Empty if no issues were encountered.
+    """
+    messages = []
+
+    wms_service_class = doImport(wms_service)
+    wms_service = wms_service_class({})
+    reports, message = wms_service.report(
+        wms_workflow_id=run_id, user=user, hist=hist, pass_thru=pass_thru, is_global=is_global
+    )
+    if message:
+        messages.append(message)
+
+    if postprocessors is not None:
```

Comment:
Also check for empty list to avoid looping for no reason?  ` if postprocessors:`